app/src/components/PlaybackControl.tsx

```diff
@@ -0,0 +1,266 @@
+     */
+    private handleUnitSpeedChange = (event: React.ChangeEvent<any>): void => {
+        const { selectedUnit } = this.state;
+        const unitSpeed: number = Number.parseFloat(event.target.value);
+        if (unitSpeed && unitSpeed < 0) {
+            this.showMessage('Please enter a positive unit # for playback.');
+        } else if (unitSpeed && selectedUnit === 'Messages' && !(Number.isInteger(unitSpeed))) {
+            this.showMessage('Please enter a whole number.');
+        } else this.setState({ unitSpeed }, () => this.checkRealTimeRate());
+    };
+
+    /**
+     * @desc Handles adding an event for starting playback and retrieving new data at specified speed
+     */
+    handlePlayOrderBook = (): void => {
+        const { playback, handlePlayback } = this.props;
+        const { unitSpeed } = this.state;
+        if (!playback) {
+            if (unitSpeed === 0) {
+                this.showMessage('0 is not a valid unit speed. Please select a strictly positive number');
+            } else {
+                handlePlayback(true);
+                this.getPlaybackOrderBookData();
+            }
+        }
+    };
+
+    /**
+     * Checks the real time rate to see whether it is valid or not.
+     * @returns {boolean}
+     */
+    checkRealTimeRate = (): boolean => {
+        let valid: boolean = true;
+        const realTimeRate = this.getRealTimeRate();
+        if (realTimeRate > MAXIMUM_PLAYBACK_REAL_TIME_RATE) {
+            this.showMessage(`Please select a lower real time rate for playback. Maximum ratio is 5. `
+             + `You entered: ${realTimeRate}`);
+            this.setState({ unitSpeed: MAXIMUM_PLAYBACK_REAL_TIME_RATE });
+            valid = false;
+        }
+        return valid;
+    };
+
+    /**
+     * @desc Gets the real time rate for playback, 0 if Messages is selected as unit
+     * @returns {number}
+     */
+    getRealTimeRate = (): number => {
+        const { unitSpeed, selectedUnit } = this.state;
+        if (selectedUnit === 'Messages') return 0;
+
+        let timeInNano: number;
+        switch (selectedUnit) {
+        case 'Seconds':
+            timeInNano = NANOSECONDS_IN_ONE_SECOND;
+            break;
+        case 'Milliseconds':
+            timeInNano = NANOSECONDS_IN_ONE_MILLISECOND;
+            break;
+        case 'Microseconds':
+            timeInNano = NANOSECONDS_IN_ONE_MICROSECOND;
+            break;
+        case 'Nanoseconds':
+            timeInNano = 1;
+            break;
+        default:
+            timeInNano = NANOSECONDS_IN_ONE_SECOND;
+        }
+        return (timeInNano / NANOSECONDS_IN_ONE_SECOND) * unitSpeed;
+    };
+
+    /**
+     * @desc Gets playback speed parameter based on selected unit and unit speed
+     */
+    getPlaybackParameter = (): string => {
+        const { unitSpeed, selectedUnit } = this.state;
+        let parameter: string = '?delay=2.5&';
```

Comment:
2.5s delay is too noticeable from a user's perspective. I tried 0.5s locally and it seems the app can handle it well. To account for this, when the user selects "Messsages" as a unit, let's multiply the input number by the delay so that they always see "X Messages per second".